rcswitch/__init__.py

```diff
@@ -0,0 +1,96 @@
+#!/usr/bin/env python3
+# vim: set encoding=utf-8 tabstop=4 softtabstop=4 shiftwidth=4 expandtab
+#########################################################################
+# Copyright 2017      Daniel Frank                knx-user-forum.de:dafra 
+#########################################################################
+# This file is part of SmartHome.py. http://mknx.github.io/smarthome/
+#
+# SmartHome.py is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# SmartHome.py is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with SmartHome.py. If not, see <http://www.gnu.org/licenses/>.
+#########################################################################
+
+import logging
+from lib.model.smartplugin import SmartPlugin
+from datetime import datetime, timedelta
+import os
+import time
+import threading
+
+class RCswitch(SmartPlugin):
+
+	ALLOW_MULTIINSTANCE = False
+	PLUGIN_VERSION = "1.2.0.1"
+
+	def __init__(self, smarthome, rcswitch_dir='/usr/local/bin/rcswitch-pi', rcswitch_sendDuration='0.5'):
+		self.logger = logging.getLogger(__name__)
+		self.logger.info('Init RCswitch')
+		self.setupOK = True
+		self._sh=smarthome
+		
+		# Check optional Parameters: check Path
+		# Step 1: format string: cut possible '/' at end of rcswitch_dir parameter
+		if rcswitch_dir[len(rcswitch_dir)-1] == '/':
+			self.rcswitch_dir = rcswitch_dir[0:len(rcswitch_dir)-1]
+		else:
+			self.rcswitch_dir = rcswitch_dir
+
+		# Step 2: check if rc switch is installed at the specified path
+		if not os.path.isfile('{}/send'.format(self.rcswitch_dir)):
+			self.logger.error('RCswitch: send file of RCswitch not found at {}. Check if RCswitch is installed correctly and path is set correctly in logic.conf.'.format(self.rcswitch_dir))
+			self.setupOK = False
+			
+		# Check optional Parameters: check sendDuration
+		try:
+			self.sendDuration = float(rcswitch_sendDuration)
+		except Exception as e:
+			self.sendDuration = float(0.5)
+			self.logger.warning('RCswitch: Argument {} for rcswitch_sendDuration is not a valid number. Using default value instead.'.format(rcswitch_sendDuration))
+			
+		# setup semaphore
+		self.lock = threading.Lock()
+		
+	def run(self):
+		self.alive = True
+
+	def stop(self):
+		self.alive = False
+
+	def parse_item(self, item):
+		# generate warnings for incomplete configured itemns
+		if 'rc_device' in item.conf:

```

Comment:
Mir waere es sogar noch lieber kein `conf` als einfaches `dict` zu haben sondern das in einem Objekt abzubilden, um flexibler zu sein. Mit den Funktionen aus dem SmartPlugin koennte eine Umstellung einfacher werden, da ein einheitliches Interface verwendet wird.

Was die Performance angeht ist das kaum bemerkbar. Abgesehen davon ist das ein Aufruf der in `parse_item()` gemacht wird, was nur beim Starten aufgerufen wird. Und ehrlich gesagt sind mir Standards lieber als die Performance. Letzteres kann man meistes durch bessere Hardware erschlagen, ersteres allerdings nur durch mehr Arbeit/Zeit (weil unterschiedliche Implementierungen beruecksichtig werden muessen) - und die Zeit ist mir heilig. :wink: